### ISSUE TITLE
Cannot run model with URL as argument

### ISSUE THREAD

**Issue by ju-w** [ultralytics/yolov5#2246]
Hello,
# Bug
#2194 broke `model.forward()` with URLs as argument and subsequently the Colab example https://colab.research.google.com/github/pytorch/pytorch.github.io/blob/master/assets/hub/ultralytics_yolov5.ipynb

# Reproduce
e.g. run Colab example.

# Trace

```
ValueError                                Traceback (most recent call last)
<ipython-input-1-86a3d667ec69> in <module>()
      9 
     10 # Inference
---> 11 results = model(imgs)
     12 
     13 # Results

2 frames
/usr/local/lib/python3.6/dist-packages/torch/nn/modules/module.py in _call_impl(self, *input, **kwargs)
    725             result = self._slow_forward(*input, **kwargs)
    726         else:
--> 727             result = self.forward(*input, **kwargs)
    728         for hook in itertools.chain(
    729                 _global_forward_hooks.values(),

/root/.cache/torch/hub/ultralytics_yolov5_master/models/common.py in forward(self, imgs, size, augment, profile)
    201             if isinstance(im, str):  # filename or uri
    202                 im = Image.open(requests.get(im, stream=True).raw if im.startswith('http') else im)  # open
--> 203             files.append(Path(im.filename).with_suffix('.jpg').name if isinstance(im, Image.Image) else f'image{i}.jpg')
    204             im = np.array(im)  # to numpy
    205             if im.shape[0] < 5:  # image in CHW

/usr/lib/python3.6/pathlib.py in with_suffix(self, suffix)
    835         name = self.name
    836         if not name:
--> 837             raise ValueError("%r has an empty name" % (self,))
    838         old_suffix = self.suffix
    839         if not old_suffix:

ValueError: PosixPath('.') has an empty name
```
# Reason
https://github.com/ultralytics/yolov5/blob/d2e754b67bc08d3634df05932cc94d8c9314a7b1/models/common.py#L201-L203
An `Image` created from `requests.get` download will not have `im.filename` property set, thus `Path(im.filename)` will fail.

**Comment by smartinellimarco** [comment]
I faced the same issue just now. Just wrote a dummy filename into the image's property.
Thanks for sharing!

**Comment by glenn-jocher** [comment]
@ju-w @MarcoCBA thanks for the bug report! Can you provide a minimum reproducible example please?
https://docs.ultralytics.com/help/minimum_reproducible_example/

**Comment by glenn-jocher** [comment]
Update, I can reproduce using the default Hub example at https://pytorch.org/hub/ultralytics_yolov5/

Will investigate, this should be working correctly but is not.

**Comment by glenn-jocher** [comment]
@ju-w @MarcoCBA this issue is fixed now in #2250. Please retry using the `force_reload=True` option when loading a new PyTorch Hub model.

Fix verified in Colab.

<img width="1270" alt="Screen Shot 2021-02-19 at 11 23 59 AM" src="https://user-images.githubusercontent.com/26833433/108551582-012f9d00-72a5-11eb-833f-29a3e075ac21.png">
